compiler/rustc_const_eval/src/interpret/projection.rs

```diff
@@ -316,7 +316,9 @@ where
     {
         use rustc_middle::mir::ProjectionElem::*;
         Ok(match proj_elem {
-            OpaqueCast(ty) => base.transmute(self.layout_of(ty)?, self)?,
+            OpaqueCast(_) => {
+                bug!("OpaqueCast should be unreachable here!");
```

Comment:
Can you make this `span_bug!(self.cur_span(), ...`? That will make this much easier to debug, should that ever come up.